asterics-web-vue/src/libraries/util/help.js

```diff
@@ -0,0 +1,90 @@
+function loadTags() {
+  return fetch("https://api.github.com/repos/asterics/AsTeRICS/tags")
+    .then(res => res.json())
+    .then(out => {
+      let r = [];
+      if (!out.hasOwnProperty("message")) {
+        out.forEach(element => {
+          if (/^v\d(.\d){0,2}$/g.test(element.name)) {
+            r.push(element.name);
+          }
+        });
+      }
+      return r;
+    })
+    .catch(err => {
+      throw err;
+    });
+}
+
+async function asyncForEach(array, cb) {
+  for (let i = 0; i < array.length; i++) [await cb(array[i], i, array)];
+}
+
+const getContent = async l => {
```

Comment:
is there a special reason for using `async` and `await`? browser compatibility is not the best and does it has any advantage?

Reply:
I'm not sure 100% of all pros and cons. I've used it currently because the code gets more readable, and so-called "callback hell" is avoided (cf. [here](https://blog.hellojs.org/asynchronous-javascript-from-callback-hell-to-async-and-await-9b9ceb63c8e8)).

---

The current project setup (vue-cli, webpack) should actually transform cases where *browser compatibility* issues could arise.

Configuration from `package.json` is used to ensure targeted browser support.

```javascript
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not ie <= 8"
  ]
```

---

I did not have enough time so far, but with following two commands, one can analyse (or read at least) the generated code, either for older browsers, or newer ones --- and verify if the generated code includes any `async`/`await` or if they are replaced by webpack or babel-loader, respectively.

- `npm run build:modern`
- `npm run build:devel`

A combination of these commands (cf. `package.json`) could be used to generate _readable_ code, generated for modern and older browsers.

Reply:
ok, until now I never used `async` and `await`, so I'm not sure either. You are right, the code maybe gets more readable, but at the same time maybe also slower, because everything is done sequential, while some things maybe could be done in parallel using callbacks and Promises.
I do not know if it is possible to make `async` and `await` compatible with older browsers at all because they are completeley new things that cannot be fixed with polyfills easily.

Yeah, but would be interesting what happens to the `async` calls in the generated code.